File: PHP/basic_api.php
<?php
/*
In PHP, we can use the standard cURL libraries to make HTTP requests. First, we define the baseURL, searchText, and bearerToken variables. Then, we construct the complete URL with the encoded search text using template literals.

Next we create a new curl instance and add the required headers. The result of the curl request is decode into an array after which the first hs_code element is printed to the console.
*/

$baseURL = "https://api.fractus-ai.com/api/search";
$searchText = "horse";
$bearerToken = "your-bearer-token";

$url = $baseURL."?q=".urlencode($searchText);

$ch = curl_init();
curl_setopt($ch,CURLOPT_URL,$url);
curl_setopt($ch,CURLOPT_CUSTOMREQUEST,"GET"); 
curl_setopt($ch,CURLOPT_RETURNTRANSFER,true);
curl_setopt($ch,CURLOPT_FOLLOWLOCATION,true);
$headers = [
    "Accept: application/json",
    "Authorization: Bearer {$bearerToken}"
];
curl_setopt($ch,CURLOPT_HTTPHEADER,$headers);
$server_output = curl_exec($ch);
curl_close($ch);

$json = json_decode($server_output,true);

print "Found HS code: {$json['data'][0]['hs_code']}\n";
?>

File: C++/basic_api.cpp
/*
In C++, we use the libcurl library to make HTTP requests. We define a callback function WriteCallback to handle the response data. Inside the main function, we initialize the necessary variables and perform the request using libcurl.

We set the URL, write callback function, and response data using curl_easy_setopt. We also set the headers using curl_slist and pass it to libcurl with CURLOPT_HTTPHEADER.

After performing the request with curl_easy_perform, we parse the serverOutput as JSON using the nlohmann/json library. Finally, we print the HS code from the parsed JSON.

Note that you need to have libcurl and nlohmann/json libraries installed and properly linked to your C++ project in order to compile and run the code.
*/

#include <iostream>
#include <curl/curl.h>
#include <nlohmann/json.hpp>
using json = nlohmann::json;

size_t WriteCallback(void* contents, size_t size, size_t nmemb, std::string* response)
{
    size_t totalSize = size * nmemb;
    response->append((char*)contents, totalSize);
    return totalSize;
}

int main()
{
    CURL* curl;
    CURLcode res;
    std::string baseURL = "https://api.fractus-ai.com/api/search";
    std::string searchText = "horse";
    std::string bearerToken = "2|tSWRRcFuIiw7spXvUIuN476I";
    std::string serverOutput;

    curl_global_init(CURL_GLOBAL_DEFAULT);
    curl = curl_easy_init();
    if (curl)
    {
        std::string url = baseURL + "?q=" + curl_easy_escape(curl, searchText.c_str(), searchText.length());
        curl_easy_setopt(curl, CURLOPT_URL, url.c_str());
        curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, WriteCallback);
        curl_easy_setopt(curl, CURLOPT_WRITEDATA, &serverOutput);
        curl_slist* headers = NULL;
        headers = curl_slist_append(headers, "Accept: application/json");
        std::string authHeader = "Authorization: Bearer " + bearerToken;
        headers = curl_slist_append(headers, authHeader.c_str());
        curl_easy_setopt(curl, CURLOPT_HTTPHEADER, headers);
        res = curl_easy_perform(curl);
        curl_slist_free_all(headers);
        curl_easy_cleanup(curl);
    }

    json json_data = json::parse(serverOutput);
    std::cout << "Found HS code: " << json_data["data"][0]["hs_code"] << std::endl;

    curl_global_cleanup();

    return 0;
}
